wremnants/histselections.py

```diff
@@ -15,39 +15,37 @@
     "ptll_mll" : "nominal",
 }
 
-def fakeHistABCD(h, low_PU=False,
-                 boolMT=True, thresholdMT=40.0, axisNameMT="mt", integrateMT=True):
+def fakeHistABCD(h, thresholdMT=40.0, fakerate_axes=["pt", "eta", "charge"], axis_name_mt="mt", integrateMT=False):
     # boolMT=False generalizes the method when mT is a real axis (i.e. no need for passMT)
```

Comment:
This new implementation looks very nice. This comment line is not needed any longer then, since whether or now there is a passMT axis to use is automatically checked using the presence of an axis named axisNameMT.

I'd like to raise a point. I think Jan had a setup where he had both the passMT axis and an actual mT axis. In this case the projection happening below to compute the fake rate should still be correct because when one integrates over passMT only the events where passMT=False will effectively be kept, since one had previously cut on mT. But this is only true if passMT was defined using the same threshold and with the same mT definition as the mT axis (usually it should be the case unless one is trying to do fancy things)

Reply:
I don't see a reason why a histogram should have both, the passMT and the mt axis, and I removed the passMT axis for the mt histogram in the lowPU histmaker.